aiven_mysql_migrate/dump_tools.py

```diff
@@ -117,3 +125,219 @@ def get_import_command(self) -> List[str]:
+        return tempfile.TemporaryDirectory()
+
+    def _get_dump_output_dir(self) -> Path:
+        """Get or create the dump output directory (subdirectory of temp_dir)."""
+        if not self.temp_dir:
+            self.temp_dir = self._create_temp_directory()
+
+        if not self.dump_output_dir:
+            self.dump_output_dir = Path(self.temp_dir.name) / "dump_output"
+            self.dump_output_dir.mkdir(parents=True, exist_ok=True)
+
+        return self.dump_output_dir
+
+    def get_dump_command(self, migration_method: MySQLMigrateMethod) -> List[str]:
+        """Build mydumper command."""
+        if not self.temp_cnf_file:
+            self.temp_cnf_file = self._create_temp_cnf_file(self.source, "source.cnf")
+
+        dump_output_dir = self._get_dump_output_dir()
+
+        assert self.temp_dir is not None, "temp_dir must exist at this point"
+
+        cmd = [
+            "mydumper",
+            f"--defaults-extra-file={self.temp_cnf_file}",
+            "--host",
+            self.source.hostname,
+            "--port",
+            str(self.source.port),
+            "--trx-tables=0",
+            "--regex",
+            r"^(?!(mysql|sys|information_schema|performance_schema)\.)",
+            "--compress=zstd",
+            "--threads=0",
+            "--triggers",
+            "--events",
+            "--routines",
+            "--chunk-filesize=1024",
+            "--sync-thread-lock-mode=FTWRL",
+            "--no-backup-locks",
+            "--skip-ddl-locks",
+            "--checksum-all",
+            "--logfile",
+            f"{self.temp_dir.name}/../mydumper.log",
+            "--verbose=4",
+            "--stream=NO_STREAM_AND_NO_DELETE",
+            "--replica-data",
+            "--source-data",
+            f"--outputdir={dump_output_dir}"
+        ]
+        if self.source.ssl:
+            cmd += ["--ssl-mode=REQUIRED"]
+        if self.databases:
+            cmd += ["--database", ",".join(shlex.quote(db) for db in self.databases)]
+
+        return cmd
+
+    def get_import_command(self, migration_method: Optional[MySQLMigrateMethod] = None) -> List[str]:
+        """Build myloader command."""
+        if not self.temp_target_cnf_file:
+            self.temp_target_cnf_file = self._create_temp_cnf_file(self.target, "target.cnf")
+
+        dump_output_dir = self._get_dump_output_dir()
+
+        assert self.temp_dir is not None, "temp_dir must exist at this point"
+
+        cmd = [
+            "myloader",
+            f"--defaults-extra-file={self.temp_target_cnf_file}",
+            "--host",
+            self.target.hostname,
+            "--port",
+            str(self.target.port),
+            "--threads=0",
+            f"--directory={dump_output_dir}",
+            "--optimize-keys=AFTER_IMPORT_ALL_TABLES",
+            "--compress-protocol=zstd",
+            "--overwrite-tables",
+            "--logfile",
+            f"{self.temp_dir.name}/../myloader.log",
```

Comment:
Same here, I don't think we should log into some temp log file, would be better to use stdout?